### PR TITLE
Overloaded AddModuleAsync with Type

### PR DESCRIPTION
**Pull request by Bond-009**
IDK

### CODE REVIEW

**Comment by MarkusG** [comment]
How is this different from `AddModule<T>()`?

**Comment by Bond-009** [comment]
You can do things like this:
```csharp
foreach (KeyValuePair<System.Type, bool> pair in dic)
{
    if (pair.Value)
    {
        await CommandService.AddModuleAsync(pair.Key);
    }
}
```

**Comment by FiniteReality** [comment]
:-1: Code duplication which could have been avoided by redirecting one method to the other. Also, there is no check that the type is actually a valid module before attempting to build it.

@MarkusGordathian `AddModule(type)` is useful for when you don't have access to use its generic twin, e.g. if you're loading modules from a source other than an Assembly.

**Comment by Bond-009** [comment]
@FiniteReality better?

File: src/Discord.Net.Commands/CommandService.cs [hunk 1]
```diff
@@ -83,20 +83,21 @@ public CommandService(CommandServiceConfig config)
                 _moduleLock.Release();
             }
         }
-        public async Task<ModuleInfo> AddModuleAsync<T>()
+        public async Task<ModuleInfo> AddModuleAsync<T>() => await AddModuleAsync(typeof(T));
```
Comment (FiniteReality) [inline src/Discord.Net.Commands/CommandService.cs]: Remove `async` and `await` here, there's no need to use an async context as we're just redirecting to the other method.

Comment (Bond-009) [reply src/Discord.Net.Commands/CommandService.cs]: Done

**Comment by FiniteReality** [comment]
:+1: LGTM, with this I can add a nice check to AddModuleAsync to prevent non-module types from attempting to be added before we try to build them.

**Review by Auralytical — CHANGES_REQUESTED** [review]
This change should be done to RemoveModuleAsync too for consistency.

File: src/Discord.Net.Commands/CommandService.cs [hunk 2]
```diff
@@ -153,13 +154,17 @@ private ModuleInfo LoadModuleInternal(ModuleInfo module)
                 _moduleLock.Release();
             }
         }
-        public async Task<bool> RemoveModuleAsync<T>()
+        public Task<bool> RemoveModuleAsync<T>()
```
Comment (FiniteReality) [inline src/Discord.Net.Commands/CommandService.cs]: Use an expression-bodied method here for consistency

**Pull request merged**